common-core-open/src/main/java/edu/isi/nlp/files/KeyValueSources.java

```diff
@@ -92,4 +93,26 @@ private KeyValueSources() {
     }
     return new ZipKeyValueSource(zipFile, ret.build());
   }
+
+  @Nonnull
+  public static ImmutableKeyValueSource<Symbol, ByteSource> fromVistaUtilsZipKeyValueSource(
+      final ZipFile zipFile) {
+    // read "__keys" file
+    ZipEntry keysEntry = zipFile.getEntry(("__keys"));
+    String keysString = null;
+    try {
+      keysString = ZipFiles.entryAsString(zipFile, keysEntry, StandardCharsets.US_ASCII);
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
In the sort of code we write, you almost never want to do this.  If there is a failure, you want it to be nice, loud, and unmistakeable. :-). in this case, you can just let the `IOException` propagate out of the method.

Reply:
oops, sounds good, thanks. this was a case of my taking the IDE's suggestion without thinking a lot about it